recipe/gen_patch_json.py

```diff
@@ -462,6 +462,13 @@ def _gen_new_index(repodata, subdir):
             if new_constrains != record.get('constrains', ()):
                 record['constrains'] = new_constrains
 
+        if record_name == "arrow-cpp":
+            if not any(dep.split(' ')[0] == "arrow-cpp-proc" for dep in record.get('constrains', ())):
+                if 'constrains' in record:
+                    record['constrains'].append("arrow-cpp-proc * cpu")
```

Comment:
Don't edit this python object as it's later checked for diff. Copy it and edit.

Reply:
https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/c15911e68d4507428b1550d7c791a0bdbe131889/recipe/gen_patch_json.py#L410 is where the copy happens, otherwise none of the other patching instructions do a copy.